app/controllers/order/client_information.js

```diff
@@ -54,38 +62,77 @@ export default Ember.Controller.extend({
     return this.get('selectedId') === 'hkId' ? 1 : 2;
   },
 
+  actionType(isOrganisation, beneficiaryId) {
+    let actionType;
+    if (isOrganisation && beneficiaryId) {
+      actionType = 'DELETE';
+    } else if (!isOrganisation && beneficiaryId) {
+      actionType = 'PUT';
+    } else if (!isOrganisation && !beneficiaryId) {
+      actionType = 'POST';
+    } else {
+      actionType = null;
+    }
+    return actionType;
+  },
+
   actions: {
     saveClientDetails(){
-      var orderId = this.get('order.id');
-      var beneficiaryId = this.get('beneficiary.id');
+      let orderParams;
+      let clientInfo = this.get('selectedPurposeId');
+      let purposeId = this.get('purposes').filterBy('identifier', clientInfo).get('firstObject.id');
+
+      const isForOrganisation = clientInfo === 'organisation';
+      orderParams = isForOrganisation ? {
+          'purpose_ids': [purposeId],
+          'beneficiary_id': null
+        } : {
+          'purpose_ids': [purposeId]
+        };
 
-      var url, actionType;
+      this.send('editOrder', orderParams, isForOrganisation);
+    },
 
-      if (beneficiaryId) {
-        url = "/beneficiaries/" + beneficiaryId;
-        actionType = "PUT";
-      } else {
-        url = "/beneficiaries";
-        actionType = "POST";
-      }
+    editOrder(orderParams, isOrganisation) {
+      let order = this.get('order');
+      let orderId = order.id;
+      let beneficiaryId = this.get('beneficiary.id');
+      let store = this.store;
 
-      let order = this.get("store").peekRecord('designation', orderId);
+      let url = beneficiaryId ? "/beneficiaries/" + beneficiaryId : "/beneficiaries";
 
-      var loadingView = getOwner(this).lookup('component:loading').append();
+      let actionType = this.actionType(isOrganisation, beneficiaryId);
 
-      new AjaxPromise(url, actionType, this.get('session.authToken'), { beneficiary: this.beneficiaryParams(), order_id: orderId })
+      let beneficiary = beneficiaryId && store.peekRecord('beneficiary', beneficiaryId);
+      let loadingView = getOwner(this).lookup('component:loading').append();
+
+      let beneficiaryParams = (["PUT", "POST"].indexOf(actionType) > -1) ?  { beneficiary: this.beneficiaryParams(), order_id: orderId } : {};
+      new AjaxPromise('/orders/' + orderId, 'PUT', this.get('session.authToken'), { order: orderParams })
         .then(data => {
-          this.get("store").pushPayload(data);
-          if(this.get("prevPath") === "client_summary") {
-            if(order) {
-              this.get("store").unloadRecord(order);
-            }
-            this.transitionToRoute('orders.client_summary', orderId);
+          store.pushPayload(data);
+          if (actionType) {
+            new AjaxPromise(url, actionType, this.get('session.authToken'), beneficiaryParams)
```

Comment:
if you run a promise with a `then` callback, make sure to return it, otherwise `loadingView.destroy();` sill run before the promise is completed

Reply:
Added `return` for both `promises`.